src/extension/devtools/devtools.ts

```diff
@@ -131,31 +134,30 @@ async function createDevtoolsPanel() {
     panelWindow = getPanelActor(window);
 
     if (!connectedToPanel) {
-      const state = devtoolsMachine.getState();
-
       panelWindow.send({
         type: "initializePanel",
-        state: state.value,
-        payload: state.context.clientContext,
+        state: devtoolsMachine.state.value,
+        payload: devtoolsMachine.state.context.clientContext,
       });
 
       panelWindow.on("retryConnection", () => {
-        devtoolsMachine.send({ type: "retry" });
+        devtoolsMachine.send("retry");
       });
 
-      devtoolsMachine.subscribe(({ state }) => {
-        panelWindow.send({ type: "devtoolsStateChanged", state: state.value });
+      const subscription = devtoolsMachine.subscribe(({ value }) => {
+        panelWindow.send({ type: "devtoolsStateChanged", state: value });
       });
+      unsubscribers.add(subscription.unsubscribe);
```

Comment:
This is the only real "change" of this PR - previously it seems that subscriptions were never unsubscribed, so I hope it's right to do that here?

Reply:
I think this might actually introduce a bug because we only subscribe the first time the panel is created. Subsequent times will skip this code block, though there are several places that call `unsubscribeFromAll` (such as when the panel is hidden) so I think you might lose the subscription as soon as you switch to a different devtools panel.

The reason why I don't actually think we need this unsubscription is because closing the devtools window actually tears this stuff down, so there should be no possibility of memory leaks. As long as devtools is open, there is only ever one subscription here.

You can test this by adding a `console.log` above this, opening devtools, "Inspect element" on the Apollo panel, close devtools, then reopen. You'll notice that the window that inspects the Apollo panel disappears. If you open up devtools again and "Inspect element" on the apollo panel, you'll only ever see a single log for the subscription.

Thats the long way of saying, I'd just leave this as-is 🙂